File: src/main/kotlin/com/lcaohoanq/sp/domains/notifications/NotificationEntity.java
package com.lcaohoanq.sp.domains.notifications;


import com.fasterxml.jackson.annotation.JsonProperty;
import com.lcaohoanq.sp.bases.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity(name = "Notifications")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NotificationEntity extends BaseEntity {

    public enum NotificationType {
        INFO,
        WARNING,
        SUCCESS,
        ERROR
    }

    public enum NotificationIcon {
        MESSAGE,
        ALERT,
        CHECK,
        ERROR
    }

    @Getter
    @AllArgsConstructor
    public enum NotificationColor {
        BLUE("#2196F3"),
        YELLOW("#FFEB3B"),
        GREEN("#4CAF50"),
        RED("#F44336");

        private final String hex;
    }


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false, unique = true)
    private Long id;

    @Enumerated(EnumType.STRING)
    private NotificationType type;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "description", columnDefinition = "TEXT", nullable = false)
    private String description;

    @Column(name = "time", nullable = false)
    private LocalDateTime time;

    @Column(name = "isRead", columnDefinition = "boolean default false")
    @JsonProperty(value = "isRead")
    private boolean read;

    @Enumerated(EnumType.STRING)
    private NotificationIcon iconName;

    @Enumerated(EnumType.STRING)
    private NotificationColor iconColorHex;

    @Column(name = "email", nullable = false)
    private String email;
}

File: src/main/kotlin/com/lcaohoanq/sp/entities/NotificationEntity.kt
package com.lcaohoanq.sp.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import com.lcaohoanq.sp.bases.BaseEntity
import com.lcaohoanq.sp.domains.user.User
import com.lcaohoanq.sp.enums.NotificationEnum
import jakarta.persistence.*

@Entity
@Table(name = "notifications")
class NotificationEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, unique = true)
    val id: Long? = null,

    @Enumerated(EnumType.STRING)
    val type: NotificationEnum.NotificationType,

    @Enumerated(EnumType.STRING)
    val iconName: NotificationEnum.NotificationIcon,

    @Enumerated(EnumType.STRING)
    val iconColorHex: NotificationEnum.NotificationColor,

    @Column(name = "title", nullable = false)
    val title: String,

    @Column(name = "description", columnDefinition = "TEXT")
    val description: String? = null,


    @Column(name = "read", nullable = false, columnDefinition = "BOOLEAN DEFAULT false")
    var read: Boolean = false,

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    val user: User? = null,



    ): BaseEntity() {
}